### DISCUSSION TITLE
When will Jedis support JDK 21, or does it already support it?

### DISCUSSION THREAD

**Discussion by kaido207** [xetorthio/jedis#3937] [Q&A]
We are considering using Jedis on JDK 21.
The README doesn't mention any guarantee of compatibility with JDK 21. 
Does Jedis already support JDK 21?
If it is planned to be supported in the future, could you please tell us the release date?

**Comment by sazzad16** [comment]
First of all thank you for using Jedis.
Secondly, thank you for creating a discussion instead of an issue. Not many users understands and/or follows it.

---

Coming to your question, please clarify what do mean by JDK 21 support. Jedis is still based on JDK 8, so ideally it should JDK 21 unless there is a breaking change and we don't have anything reported so far.

About virtual thread compatibility, [check here](https://github.com/redis/jedis/discussions/3879)

**Reply by kaido207** [reply]
Thank you for your quick reply.

I am aware of the backwards compatibility of the JDK, and I understand that the current Jedis will work fine with JDK21.
However, the current README's JDK Compatibility lists 8, 11, and 17, so it doesn't seem like clearly compatible that it will work with JDK21.
I suggest changing the description to 1 or 2 as follows:
1. Add JDK21 to the list.
2. Add description that it is based on JDK8 and is supported to work with later JDKs.

This change makes it clear that Jedis works with JDK21 so that users can use it without confusion.

**Reply by sazzad16** [reply]
Good point. Thank you for raising this.

**Reply by kaido207** [reply]
Thank you for your support.
